### PR TITLE
Some cleanings viewer3d

### PR DESCRIPTION
**Pull request by kerautret**
Some cleanings and improve in viewer3d

### CODE REVIEW

File: src/DGtal/io/viewers/Viewer3D.h [hunk 1]
```diff
@@ -1028,6 +1028,135 @@ namespace DGtal
...
+     **/
+    void glCreateListPolygonsWired(const std::vector<VectorPolygon>  &aVectPolygon, unsigned int idList);
+
+    
+    /**
+     * Update the container of GLTextureImage object with the given vector of TextureImage. 
+     * @param[in] aVectImage the vector containing 
+     *
+     **/    
+    void glUpdateTextureImages(const std::vector<Viewer3D<Space, KSpace>::TextureImage>  &aVectImage);
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/io/viewers/Viewer3D.h]: Perhaps Create also a type VectorTextureImage

File: src/DGtal/io/viewers/Viewer3D.ih [hunk 2]
```diff
@@ -182,10 +182,18 @@ DGtal::Viewer3D< Space ,KSpace >::updateTextureImage(unsigned int imageIndex, co
   BOOST_CONCEPT_ASSERT(( concepts::CConstImage < TImageType > ));
   assert ( imageIndex< myGSImageList.size());
   typename Viewer3D< Space ,KSpace >::TextureImage &anImage = myGSImageList.at(imageIndex);
-  Display3D< Space, KSpace>::updateBoundingBox(DGtal::Z3i::RealPoint(anImage.point1[0]+xTranslation, anImage.point1[1]+yTranslation, anImage.point1[2]+zTranslation));
-  Display3D< Space, KSpace>::updateBoundingBox(DGtal::Z3i::RealPoint(anImage.point2[0]+xTranslation, anImage.point2[1]+yTranslation, anImage.point2[2]+zTranslation));
-  Display3D< Space, KSpace>::updateBoundingBox(DGtal::Z3i::RealPoint(anImage.point3[0]+xTranslation, anImage.point3[1]+yTranslation, anImage.point3[2]+zTranslation));
-  Display3D< Space, KSpace>::updateBoundingBox(DGtal::Z3i::RealPoint(anImage.point4[0]+xTranslation, anImage.point4[1]+yTranslation, anImage.point4[2]+zTranslation));
+  Display3D< Space, KSpace>::updateBoundingBox(DGtal::Z3i::RealPoint(anImage.point1[0]+xTranslation,
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/io/viewers/Viewer3D.ih]: Perhaps define inner types
typedef Display3D< Space, KSpace> Display;
typedef DGtal::Z3i::RealPoint RealPoint;

File: src/DGtal/io/viewers/Viewer3D.ih [hunk 3]
```diff
@@ -623,22 +638,23 @@ DGtal::Viewer3D<Space, KSpace>::drawWithNames()
   // selection is left clic + shift key.
   // JOL: 2014/10/15. This is my addition for interacting with
   // quads. Seems to work well.
-  unsigned int nbListOfPrimitives = Viewer3D<Space, KSpace>::myLineSetList.size() +Viewer3D<Space, KSpace>::myCubeSetList.size()+ Viewer3D<Space, KSpace>::myBallSetList.size();
-  glCallList ( GLuint ( myListToAff + nbListOfPrimitives + 1 ) );
+  //  unsigned int nbListOfPrimitives = Viewer3D<Space, KSpace>::myLineSetList.size() +Viewer3D<Space, KSpace>::myCubeSetList.size()+ Viewer3D<Space, KSpace>::myBallSetList.size();
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/io/viewers/Viewer3D.ih]: Dead code ?

File: src/DGtal/io/Display3D.h [hunk 4]
```diff
@@ -424,7 +424,7 @@ namespace DGtal
      * @return the select callback function that match the given \a
      * name, or 0 if none is associated to this name.
      */
-    SelectCallbackFct getSelectCallback3D( DGtal::int32_t name, void*& data ) const;
+    SelectCallbackFct getSelectCallback3D( DGtal::int32_t aName, void*& data ) const;
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/io/Display3D.h]: Need to update the comment above "name" => "aName".
That's why clang fail on travis.

File: src/DGtal/io/Display3DFactory.h [hunk 5]
```diff
@@ -613,7 +613,7 @@ namespace DGtal
      * @param name3d an object storing the "OpenGL" name.
      */
     static void 
-    draw( Display & display, const DGtal::SetName3D& name3d );
+    draw( Display & display, const DGtal::SetName3D& aName3d );
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/io/Display3DFactory.h]: Same comment as above. The comment above should be update "name3d" => "aName3d'.

**Comment by JacquesOlivierLachaud** [comment]
Examples work on my computer. However, the selection (shift+click) of a digital point in some viewer examples (viewer-[2-5]* at least) introduce a seg fault. I think it is because of some former code on set selection and removal.

**Comment by JacquesOlivierLachaud** [comment]
It is much better now to have separate all the codes to generate GL lists.

File: src/DGtal/io/viewers/Viewer3D.h (hunk 1 above)
Comment (kerautret) [inline src/DGtal/io/viewers/Viewer3D.h]: agree thank, done

File: src/DGtal/io/viewers/Viewer3D.ih (hunk 2 above)
Comment (kerautret) [inline src/DGtal/io/viewers/Viewer3D.ih]: Yes, I add it for Display and also for RealPoint but I change with the one of Viewer3D:
(defined as Space::RealPoint), so I just I switch all  DGtal::Z3i::RealPoint into RealPoint with def from Viewer3d by using:
    typedef typename Display::RealPoint RealPoint;

File: src/DGtal/io/viewers/Viewer3D.ih (hunk 3 above)
Comment (kerautret) [inline src/DGtal/io/viewers/Viewer3D.ih]: yes removed

File: src/DGtal/io/Display3D.h (hunk 4 above)
Comment (kerautret) [inline src/DGtal/io/Display3D.h]: Yes Thank , I miss it! ;)

File: src/DGtal/io/Display3DFactory.h (hunk 5 above)
Comment (kerautret) [inline src/DGtal/io/Display3DFactory.h]: done

**Comment by kerautret** [comment]
I see also the set fault, I look it thank

**Comment by kerautret** [comment]
I fix it, it was in fact in getSelectCallback3D, I add a test   if (it_up == mySelectCallBackFcts.begin()){
to check if only one data/func are found.
But by the way, why is the second look look needed ? from line 174 of new Display3D.ih ?

**Comment by kerautret** [comment]
I check the remaining warnings

**Comment by JacquesOlivierLachaud** [comment]
In fact, we need to look for a match twice because we are looking not for an exact, but for a range of values. Hence the find is done with a call to "upper_bound". Then a correct answer is either the returned iterator or the previous value, since ranges have no overlap. (All this is said "as far as I remember").

**Comment by JacquesOlivierLachaud** [comment]
It seems it is no more Travis that is running continued integration. For the error returned by AppVeyor, I guess @dcoeurjo is the best to debunk it.

**Comment by kerautret** [comment]
Ha ok I see :)
yes it looks fine for travis no more any warnings ;)
Thank you

**Comment by dcoeurjo** [comment]
Yeah, sorry for appveyor, I'm debugging it but it's not ready (supposed to be a Travis like for MSwindows VM)

**Comment by kerautret** [comment]
ok so perhaps we can merge?

**Comment by dcoeurjo** [comment]
Appveyor disabled. 
BTW, it may not be related but could you have a look to #943 ? this issue is related to recent refactoring you did on display3D

**Comment by kerautret** [comment]
ok I look it

**Comment by JacquesOlivierLachaud** [comment]
I wait for the answer to David's question before merging:
"As far as I remember, lists were used in board3d with the following semantic: objects within a list share the same material (to reduce MTL filesize). Now, it may be deprecated since I use now a std::map for material.

Anyway, what is the semantic you attach to lists in your recent PR?"

What I did for surfel interaction, is related to assigning a name to surfel(s), but it's true I don't know how this conflicts with lists. I guess we should say that all elements of a list have the same opengl name.

**Comment by kerautret** [comment]
yes I answered, we can perhaps keep only map, the pb that can appears is perhaps the SetName3D( name++ ) which can overwrite an other list generated by CustomColor for instance.

**Comment by kerautret** [comment]
I correct the last remarks with the fix avoiding the user list creation and an color add in testBoard3D with quad.

**Comment by dcoeurjo** [comment]
back on this topic (sorry for being slow on that): beside list usage in Board3D and Viewer3D, can we clarify the list thing and say something like that:
   a list is a collection of objectswhich are actionable

  - Viewer3D:  objects in list have the same OpenGL ID and thus can be "selected" (current situation)
  - Board3D: lists are exported with the notion of OBJ groups meaning that they can be "selected" in a modeler (e.g. blender)
  - Board3Dto2D: no consequence

If we agree, than I should remove my previous usage that objects in a list share the same material (only for perf issues not needed anymore), and remove the fact that changing the color (or mode) create a new list (current behavior).
 Maybe, @kerautret, you should check that being "actionable" is the only usage of list you have in Viewer3D, i don' t know

Would you agree with that?

**Comment by JacquesOlivierLachaud** [comment]
Seems reasonnable to me. I just do not know if an OpenGL list can have different materials. Bertrand, do you know more ?

**Comment by kerautret** [comment]
@dcoeurjo @JacquesOlivierLachaud Yes I confirm the list are not exploited in the spirit of material.
So I am ok for the proposition. Perhaps we merge first this one and then list update ? 
Or else you can PR it on my branch for the review.

**Comment by JacquesOlivierLachaud** [comment]
I suggest to merge and to do another PR afterwards. Ok for you David @dcoeurjo ?

**Comment by dcoeurjo** [comment]
Perfect :)

**Comment by dcoeurjo** [comment]
Done.. Thanks

**Pull request merged**